File: 240116/키를 기준으로 정렬/sort-by-height.cpp
#include <iostream>
#include <algorithm>
#include <string>

using namespace std;

class Student {
    public:
        string name;
        int height, weight;

        Student(string name, int height, int weight) {
            this->name = name;
            this->height = height;
            this->weight = weight;
        }

        Student(){}
};

bool cmp(Student a, Student b) { // 국어 점수 기준 내림차순 정렬
    return a.height < b.height; 
}

int main() {
    int a;
    cin >> a;
    Student students[a];

    for(int i = 0; i < a; i++){
        string b;
        int c, d;
        cin >> b >> c >> d;
        students[i] = Student(b,c,d);
    }


    sort(students, students + a, cmp);

    for(int i = 0; i < a; i++)
        cout << students[i].name << " " << students[i].height << " " << students[i].weight << endl;

    return 0;
}

File: 240603/키를 기준으로 정렬/sort-by-height.js
let fs = require("fs");
let input = fs.readFileSync(0).toString().split("\n");
let n = Number(input[0]);

class User{
    constructor(name , height , weight){
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
}
let users = [];
for(let i = 1; i <= n; i++){
    let [name , height , weight] = input[i].split(" ");
    users.push(new User(name , Number(height) , Number(weight)));
}
users.sort((a , b) => a.height - b.height);
for(let s of users){
    console.log(`${s.name} ${s.height} ${s.weight}`)
}